src/number.rs

```diff
@@ -710,6 +711,22 @@ impl Default for INumber {
     }
 }
 
+impl<A: DefragAllocator> Defrag<A> for INumber {
+    fn defrag(mut self, defrag_allocator: &mut A) -> Self {
+        let hd = self.header();
+        if hd.type_ == NumberType::Static {
+            return self;
+        }
+        unsafe {
+            let ptr = self.0.ptr().cast::<Header>();
+            self.0.set_ptr(
+                defrag_allocator.realloc_ptr(ptr.cast(), Self::layout((*ptr).type_).unwrap()),
+            )
+        };
+        self
+    }
+}
```

Comment:
Three comments:
Use `expect` instead of `unwrap`.
Would it be possible to get the `ptr` from the `Header` itself instead of going through the `IValue::ptr`?
And would it be possible to reuse the already extracted `type` instead of accessing it again through `ptr`?

eg something like:
```suggestion
impl<A: DefragAllocator> Defrag<A> for INumber {
    fn defrag(mut self, defrag_allocator: &mut A) -> Self {
        let hd = self.header();
        match hd.type_ {
            NumberType::Static => self,
            t => {
                let ptr = hd.ptr();
                let layout = Self::layout(t).expect(...);
                unsafe {
                    self.0.set_ptr(defrag_allocator.realloc_ptr(ptr.cast(), layout));
                }
                self
            }
        }
    }
}
```